app/models/job.rb

```diff
@@ -20,4 +20,10 @@ class Job < ActiveRecord::Base
             presence: true,
             inclusion: { in: STATES }
   validates :author, presence: true
+
+  def self.search(search)
```

Comment:
I forgot about that comment. 

You can use a named scope for that. Named scopes are pretty useful to chain ActiveRecord queries. You wont have to use a `base_relation` variable (but I like your usage of a guard clause).